Support global WakaTime installation

notquiteamonad opened the pull request:
This is quick-and-dirty for the moment but I'll clean it up if you think it's worth adding as an option. For systems like NixOS ones where the paths are unpredictable, I added a dependency on node-which. This could also replace or be used alongside the current search for python, although the existing method found my python executable fine.

Feel free to flat-out reject this if it goes against what you're aiming for in terms of config surface etc.

Closes #177.

jakeisnt commented:
I need this to get NixOS working - would love to see this merged.

jakeisnt commented:
Unfortunately, installing python37 doesn't do it for me - all my extensions [are installed declaratively](https://github.com/jakeisnt/nix-cfg/blob/vim/modules/editors/vscode.nix), and because they're saved as Nix derivations, the extension can't pull down the `wakatime-cli` python file due to the Nix store being read-only. Thanks for the PR, though; I hope this can be merged soon!

notquiteamonad commented:
That link's into a private repo unfortunately so I can't see it - although if you'd be willing to publish it as a gist or something I'd be interested to see how you're doing that. Working on cleaning this up now :slightly_smiling_face:

jakeisnt commented:
I just gave you the wrong link - try this! https://github.com/jakeisnt/nix-cfg/blob/master/modules/editors/vscode.nix

notquiteamonad commented:
I've rebased onto master and changed a couple of fatal logs into errors to quit out if there's no global installation. It works for me using a global or non-global installation with `vscodium` package from the unstable NixOS channel (this extension requires version 52 or greater, not available on 20.09). Anything else you want changing I'll be happy to do so.

jakeisnt commented:
Remove 'WIP' from the title and request a review!

notquiteamonad commented:
Doh! Thanks :joy:

notquiteamonad commented:
@jakeisnt I've followed your example and configured all my VSCode extensions in my nixos config as well now. Although I'm going to have to switch back to manual installs until this is merged as you can't mix the two :sob: 

@alanhamlett is there any chance you could take a look at this?

jakeisnt commented:
Would it be possible to get a release candidate for this?

alanhamlett commented:
Released in v5.0.1 today.

notquiteamonad commented:
Thank you ☺️
